PiwikTracker/Tracker.swift

```diff
@@ -63,4 +65,60 @@ final public class Tracker: NSObject {
         }
         self.dispatchTimer = Timer.scheduledTimer(timeInterval: dispatchInterval, target: self, selector: #selector(dispatch), userInfo: nil, repeats: false)
     }
+    
+    var visitor = Visitor.current()
+    var session = Session.current()
+}
+
+// shared instance
+extension Tracker {
+    public static var sharedInstance: Tracker? {
+        get { return _sharedInstance }
+    }
+    
+    public class func configureSharedInstance(withSiteID siteID: String, baseURL: URL) {
+        let queue = MemoryQueue()
+        let dispatcher = URLSessionDispatcher(baseURL: baseURL)
+        self._sharedInstance = Tracker.init(siteId: siteID, queue: queue, dispatcher: dispatcher)
+    }
+    
+    public class func configureSharedInstance(withSiteID siteID: String, queue: Queue = MemoryQueue(), dispatcher: Dispatcher) {
+        self._sharedInstance = Tracker(siteId: siteID, queue: queue, dispatcher: dispatcher)
+    }
+}
+
+extension Tracker {
+    func event(action: [String]) -> Event {
+        return Event(
+            siteId: siteId,
+            uuid: NSUUID(),
+            visitor: visitor,
+            session: session,
+            date: Date(),
+            url: URL(string: "http://example.com")!.appendingPathComponent(action.joined(separator: "/")),
+            actionName: action,
+            language: Locale.httpAcceptLanguage,
+            isNewSession: false, // set this to true once we can start a new session
+            referer: nil)
+    }
+}
+
+extension Locale {
+    static var httpAcceptLanguage: String {
+        var components: [String] = []
+        for (index, languageCode) in preferredLanguages.enumerated() {
+            let quality = 1.0 - (Double(index) * 0.1)
+            components.append("\(languageCode);q=\(quality)")
+            if quality <= 0.5 {
+                break
+            }
+        }
+        return components.joined(separator: ",")
+    }
+}
+
+extension Tracker {
+    public func track(view: [String]) {
```

Comment:
I actually is not tracking multiple views. A view can be described by a path/categories. For example videos/cats/15. 

Your question definitely shows an issue. All public methods, especially the Tracker, should be documented.

- [ ] Document public Tracker methods.